### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by mwhamgenomics**
Refactoring before starting work on #222.

### CODE REVIEW

File: analysis_driver/quality_control/contamination_blast.py [hunk 1]
```diff
@@ -1,128 +0,0 @@
-import json
```
Comment (mwhamgenomics) [inline analysis_driver/quality_control/contamination_blast.py:1]: Moved to `contamination_checks` and renamed to `Blast`

File: analysis_driver/quality_control/gender_validation.py [hunk 2]
```diff
@@ -39,24 +37,3 @@ def _run(self):
...
-
-def _parse_args():
-    p = argparse.ArgumentParser()
-    p.add_argument('--sample_id', type=str, help='sample ID for creating a Sample dataset object')
-    p.add_argument('-v', '--vcf_file', dest='vcf_file', type=str, help='the vcf file used to detect the gender')
-    p.add_argument('-s', '--working_dir', dest='working_dir', type=str, help='the working dir for execution')
-    return p.parse_args()
-
-if __name__ == '__main__':
-    sys.exit(main())
```
Comment (mwhamgenomics) [inline analysis_driver/quality_control/gender_validation.py:62]: This was already in `run_qc`, so removed.

File: analysis_driver/reader/mapping_stats_parsers.py [hunk 3]
```diff
@@ -6,27 +6,6 @@
...
-            sp_line = line.strip().split()
-            if line.startswith('Total reads:'):
-                total_reads = sp_line[2]
-            elif line.startswith('Mapped reads:'):
-                mapped_reads = sp_line[2]
-            elif line.startswith('Duplicates:'):
-                duplicate_reads = sp_line[1]
-            elif line.startswith("'Proper-pairs':"):
-                proper_pairs = sp_line[1]
-    return int(total_reads), int(mapped_reads), int(duplicate_reads), int(proper_pairs)
```
Comment (mwhamgenomics) [inline analysis_driver/reader/mapping_stats_parsers.py:27]: We should have removed this in #225

File: analysis_driver/reader/sample_sheet.py [hunk 4]
```diff
@@ -1,180 +0,0 @@
-import csv
```
Comment (mwhamgenomics) [inline analysis_driver/reader/sample_sheet.py:1]: We don't need this anymore, can reinstate it if we want to be able to process runs with handwritten sample sheets while the Lims is down.

File: analysis_driver/report_generation/run_crawler.py [hunk 5]
```diff
@@ -287,136 +250,13 @@ def _populate_barcode_info_from_conversion_file(self, conversion_xml):
...
+            (
+                pp('run_elements', self.barcodes_info.values(), ELEMENT_RUN_ELEMENT_ID),
+                pp('unexpected_barcodes', self.unexpected_barcodes.values(), ELEMENT_RUN_ELEMENT_ID),
+                pp('lanes', self.lanes.values(), ELEMENT_LANE_ID),
+                pp('runs', [self.run], ELEMENT_RUN_NAME),
+                pp('samples', self.libraries.values(), ELEMENT_SAMPLE_INTERNAL_ID, ['run_elements']),
+                pp('projects', self.projects.values(), ELEMENT_PROJECT_ID, ['samples'])
             )
-
-
```
Comment (mwhamgenomics) [inline analysis_driver/report_generation/run_crawler.py:302]: Split off into `report_generation.sample_crawler`

File: analysis_driver/transfer_data.py [hunk 6]
```diff
@@ -1,33 +1,11 @@
 import os
 from egcg_core import clarity, util, archive_management
-from analysis_driver.config import default as cfg
 from egcg_core.app_logging import logging_default as log_cfg
-from egcg_core.constants import ELEMENT_RUN_NAME, ELEMENT_LANE, ELEMENT_PROJECT_ID, ELEMENT_NB_READS_CLEANED
 
 app_logger = log_cfg.get_logger(__name__)
 
 
-def prepare_sample_data(dataset):
```
Comment (mwhamgenomics) [inline analysis_driver/transfer_data.py:10]: Moved to `pipelines.common.MergeFastqs`

File: analysis_driver/util/__init__.py [hunk 7]
```diff
@@ -1,77 +1,13 @@
...
-from egcg_core.util import str_join
-
-from analysis_driver.config import default as cfg
 from analysis_driver.reader.run_info import Reads
 from . import bash_commands
 
 app_logger = log_cfg.get_logger('util')
 
 
-def bcbio_prepare_samples_cmd(job_dir, sample_id, fastqs, user_sample_id):
```
Comment (mwhamgenomics) [inline analysis_driver/util/__init__.py:19]: Moved to `pipelines.common.MergeFastqs`

File: analysis_driver/util/__init__.py [hunk 8]
```diff
@@ -1,77 +1,13 @@
...
-
-    with open(csv_file, 'w') as f:
-        writer = csv.writer(f)
-        writer.writerow(['samplename', 'description'])
-        for fq in fastqs:
-            writer.writerow([fq, user_sample_id])
-
-    return csv_file
-
-
```
Comment (mwhamgenomics) [inline analysis_driver/util/__init__.py:55]: Moved to RunDataset

File: integration_tests/integration_test.py [hunk 9]
```diff
@@ -28,72 +27,6 @@ def _now():
     return datetime.utcnow().strftime('%d/%m/%Y %H:%M:%S')
 
 
-def _fake_get_list_of_samples(sample_names):
```
Comment (mwhamgenomics) [inline integration_tests/integration_test.py:31]: Moved to `integration_tests.mocked_data`

File: tests/test_quality_control/test_contamination_blast.py [hunk 10]
```diff
@@ -1,192 +0,0 @@
-import os
```
Comment (mwhamgenomics) [inline tests/test_quality_control/test_contamination_blast.py:1]: Moved to `test_contamination_checks`

File: tests/test_util.py [hunk 11]
```diff
@@ -1,142 +1,54 @@
...
-from analysis_driver.util import get_ranges, convert_bad_cycle_in_trim
-from tests.test_analysisdriver import TestAnalysisDriver
-from analysis_driver import transfer_data, util
-from analysis_driver.config import OutputFileConfiguration
-
-
-def patched_find_project_from_sample(sample_id):
-    return patch('egcg_core.clarity.find_project_name_from_sample', return_value='proj_' + sample_id)
-
-
```
Comment (mwhamgenomics) [inline tests/test_util.py:14]: Moved to `test_transfer_data`

File: README.md [hunk 12]
```diff
@@ -4,205 +4,145 @@
...
-
-#### demultiplexing_parsers
-Functions for picking up QC information from `ConversionStats.xml` and `demultiplexing_stats.xml` (generated
-by bcl2fastq) and also files produced by seqtk, containing base counts with adaptors removed.
+- run_info
+  - RunInfo - Represents an instance of RunInfo.xml. Reads in the xml and constructs a Mask object from the
+    `<Reads>` element.
+  - Mask - Represents a read mask to be passed to bcl2fastq via sample_sheet.generate_mask.
+- mapping_stats_parsers - Functions that pick up QC information from files generated by BCBio, including
+  `bamtools_stats.txt`, `sort-callable.bed`, `highdepth-stats.yaml` and fastqc reports.
```
Comment (tcezard) [inline README.md]: That should be `samtools_stats.txt`

**Review by tcezard — APPROVED** [review]
This looks good

**Pull request merged**